packages/fewer/src/Association/index.ts

```diff
@@ -19,19 +21,24 @@ export enum AssociationType {
 const BASE_TYPE = Symbol('base-type');
 const FK_TYPE = Symbol('fk-type');
 
+type CheckUsedKeys<T, K> = T extends K ? 'This key is already in use.' : T;
+
 export class Association<
   Type extends AssociationType = any,
   Base extends Repository = any,
   FK = any,
   RepoType = any,
   SelectionSet = any,
   LoadAssociations extends Associations = {},
-  JoinAssociations extends Associations = {}
+  JoinAssociations extends Associations = {},
+  Chained = any,
+  SchemaType = {},
 > implements CommonQuery<RepoType, LoadAssociations & JoinAssociations> {
   // NOTE: We need to stash the type here otherwise the generic won't become a constraint.
   readonly [BASE_TYPE]: Base;
   readonly [FK_TYPE]: FK;
 
+  readonly [INTERNAL_TYPES.IS_CHAINED]: Chained;
```

Comment:
Does this need to be in INTERNAL_TYPES, or can it just be stashed like FK and Base to create the constraint?